torch/csrc/jit/python/script_init.cpp

```diff
@@ -459,9 +459,7 @@ static TypePtr inferShapeAndTypeForInput(
     }
     return TupleType::create(types);
   } else if (auto list_type = input_type->cast<ListType>()) {
-    const TypePtr& sub_type = list_type->getElementType();
-    auto elem_type =
-        inferShapeAndTypeForInput(sub_type, s_iter, s_iter_end, complete);
+    const TypePtr& elem_type = list_type->getElementType();
```

Comment:
Previous code recursively calls itself, it should take same effect as this change if `list_type->getElementType()` hits
```
  } else {
    // Primitive type, keep as is.
    s_iter++;
    return input_type;
  }
```

but for others it's different. Even for tensor type
```
    auto type = getTensorType(s_iter->toTensor(), complete);
    s_iter++;
    return type;
```

Reply:
Find aten::pad_sequence schema